flickit-assessment-kit/src/main/java/org/flickit/assessment/kit/adapter/out/persistence/kitversion/KitVersionPersistenceJpaAdapter.java

```diff
@@ -38,14 +38,15 @@ public long persist(CreateKitVersionPort.Param param) {
         var kitEntity = kitRepository.findById(param.kitId())
             .orElseThrow(() -> new ResourceNotFoundException(KIT_ID_NOT_FOUND));
 
-        var versionEntity = KitVersionMapper.createParamToJpaEntity(kitEntity, param);
-        versionEntity.setId(sequenceGenerators.generateKitVersionId());
+        var id = sequenceGenerators.generateKitVersionId();
+        var versionEntity = KitVersionMapper.createParamToJpaEntity(id, kitEntity, param);
         return repository.save(versionEntity).getId();
     }
 
     @Override
     public void updateStatus(long kitVersionId, KitVersionStatus newStatus) {
-        repository.updateStatus(kitVersionId, newStatus.getId());
+        long statusVersion = newStatus == KitVersionStatus.ARCHIVE ? -kitVersionId : newStatus.getId();
+        repository.updateStatus(kitVersionId, newStatus.getId(), statusVersion);
```

Comment:
**Suggestion:** Refactor the `updateStatus` method to handle the generation of `statusVersion` more transparently and safely, encapsulating the logic within a dedicated method. [general, importance: 8]
```suggestion
        long statusVersion = KitVersionMapper.generateStatusVersion(newStatus, kitVersionId);
        repository.updateStatus(kitVersionId, newStatus.getId(), statusVersion);
```